main.py

```diff
@@ -69,23 +86,91 @@ def draw() -> Shape:
     return target_shape
 
 
+def draw_second_stage() -> Shape:
+    """Draw the next round screen and return the chosen shape"""
+    pygame.draw.rect(SCREEN, (0, 0, 0), (0, 0, 800, 600))
+
+    random.shuffle(shapes)
+    for index, shape in enumerate(shapes):
+        if index > 0:
+            break
+
+        position = shape.map(positions[index])
+
+        # TODO: Fix the issue with pygame and being able to draw the shape from the shape class
+        # TODO: This is a dirty hack because of the 24 hour deadline from start to finish
+        if shape.method == 'rect':
+            pygame.draw.rect(SCREEN, shape.colour, position)
+        elif shape.method == 'circle':
+            pygame.draw.circle(SCREEN, shape.colour, position, shape.radius)
+        elif shape.method == 'polygon':
+            pygame.draw.polygon(SCREEN, shape.colour, position)
+        elif shape.method == 'line':
+            pygame.draw.line(SCREEN, shape.colour, position[0], position[1], shape.radius)
+
+    target_shape = shapes[random.randint(0, 3)]
+    while target_shape.method == last_shape:
+        target_shape = shapes[random.randint(0, 3)]
+
+    font = pygame.font.Font(None, 48)
+    text = font.render(f"{current_round}. Please select the {target_shape.type}", True, (255, 255, 255))
+    text_rect = text.get_rect(center=(400, 50))
+    SCREEN.blit(text, text_rect)
+
+    return target_shape
+
+
+def draw_third_stage() -> Shape:
+    """Draw the next round screen and return the chosen shape"""
+    pygame.draw.rect(SCREEN, (0, 0, 0), (0, 0, 800, 600))
+    path = 'D:\pedro\Documents\Tese\data_collect_py\IAPS/*.jpg'
+    IAPS = []
+    for filename in glob.glob(path):
+        IAPS.append(filename)
+    figure = random.choice(IAPS)
+    if figure in picked_IAPS:
```

Comment:
As mentioned above, picked_IAPS is always empty, so this code is not doing anything at the moment unfortunately.

Reply:
I can I prevent the images from repeating?

Reply:
@pedrorribeiro Yep, an easy way of preventing images from repeating is to `image = IAPS.pop(filename)`, this will define image as the selected image but remove it from the list entirely for the next time you grab a random image.